packages/core-sdk/src/resources/group.ts

```diff
@@ -388,7 +395,78 @@ export class GroupClient {
       handleError(error, "Failed to register group and attach license and add ips");
     }
   }
+  /**
+   * Collect royalties for the entire group and distribute the rewards to each member IP's royalty vault.
+   *
+   * Emits an on-chain {@link https://github.com/storyprotocol/protocol-core-v1/blob/v1.3.1/contracts/interfaces/modules/grouping/IGroupingModule.sol#L38 | `CollectedRoyaltiesToGroupPool`} event.
+   */
+  public async collectAndDistributeGroupRoyalties({
+    groupIpId,
+    currencyTokens,
+    memberIpIds,
+    txOptions,
+  }: CollectAndDistributeGroupRoyaltiesRequest): Promise<CollectAndDistributeGroupRoyaltiesResponse> {
+    try {
+      const object: GroupingWorkflowsCollectRoyaltiesAndClaimRewardRequest = {
+        groupIpId: validateAddress(groupIpId),
+        currencyTokens: validateAddresses(currencyTokens),
+        memberIpIds: validateAddresses(memberIpIds),
+      };
+      if (!object.currencyTokens.length) {
+        throw new Error("At least one currency token is required.");
+      }
+      if (!object.memberIpIds.length) {
+        throw new Error("At least one member IP ID is required.");
+      }
+      if (object.currencyTokens.some((token) => token === zeroAddress)) {
+        throw new Error("Currency token cannot be the zero address.");
+      }
+      const isGroupRegistered = await this.ipAssetRegistryClient.isRegistered({
+        id: object.groupIpId,
+      });
+      if (!isGroupRegistered) {
+        throw new Error(`The group IP with ID ${object.groupIpId} is not registered.`);
+      }
+      await Promise.all(
+        object.memberIpIds.map(async (ipId) => {
+          const isMemberRegistered = await this.ipAssetRegistryClient.isRegistered({
```

Comment:
Note: not needed to be addressed in this PR, just wanted to start a discussion.

I'm wondering if its actually worth checking for ip being registered? Each check will be an RPC call that slows things down and adds additional traffic to the RPC. I think its also very rare for clients to pass an invalid ip. Even if they do, the protocol call will just fail.
Thoughts? @bpolania @edisonz0718 @bonnie57

Reply:
I think if protocol returns clear error message or error code, we can just use protocol's error return

Reply:
@bonnie57 can you confirm if the protocol error is clear?